[SYCLomatic][DOC] Add more manual fix examples in the document (#1230)
oneapi-src/SYCLomatic: docs/dev_guide/diagnostic_ref/dpct1056.rst

Suggestions to Fix
------------------

Command:

.. code-block:: bash
   :linenos:

   dpct --out-root out test.c --extra-arg=-xc

If this variable is only used in host code, ignore this warning.

For example, this original CUDA\* code:

.. code-block:: cpp
   :linenos:

>    // dpct --out-root out test.c --extra-arg=-xc

Reviewer: ```suggestion

```

Author: done

   
   // test.c 
   #include <stdio.h>
   #include <cuda_runtime.h>
   
   static __constant__ const float aaa = 10.f;
   

>    void foo(){

Reviewer: ```suggestion
   void foo_host(){

```

Author: done

     printf("%f\n", aaa);
   }

results in the following migrated SYCL\* code:

.. code-block:: cpp
   :linenos:

   /*
   DPCT1056:0: The use of aaa in device code was not detected. If this variable is
   also used in device code, you need to rewrite the code.
   */
   static const float aaa = 10.f;
   
   void foo(){
     printf("%f\n", aaa);
   }

which is rewritten to:

.. code-block:: cpp
   :linenos:

   static const float aaa = 10.f;
   
   void foo(){
     printf("%f\n", aaa);
   }

> If the variable is also used in device code,

Reviewer: re-org the section "which is rewritten to" by adding 2 if-then blocks.

Author: done

Suggestions to Fix
------------------

> Command:

Reviewer 2: If this sample command is the command use to migrate the example CUDA code to the example migrated SYCL code, it needs introduction/context.  If there is anything of note about the command, it should be explained. For example: 

"For example, this original CUDA code: .."
"is migrated using the following migration command: [example command] "
"which results in the following migrated SYCL code: ..."
"If the variable "const_a" is only used in host code, just ignore the DPCT1056 warning. For example: .."
"If the variable "const_a" is used in both host and device code, rewrite the migrated code. For example: ... "

Reviewer: addressed

.. code-block:: bash
   :linenos:

   dpct --out-root out test.h --extra-arg=-xc

For example, this original CUDA\* code:

.. code-block:: cpp
   :linenos:

   #include <stdio.h>
   #include <cuda_runtime.h>
   

>    static __constant__ const float aaa = 10.f;

Author: ```suggestion
   static __constant__ const float const_a = 10.f;

```

Suggestions to Fix
------------------

Command:

.. code-block:: bash
   :linenos:

   dpct --out-root out test.h --extra-arg=-xc

For example, this original CUDA\* code:

.. code-block:: cpp
   :linenos:

   #include <stdio.h>
   #include <cuda_runtime.h>
   
   static __constant__ const float aaa = 10.f;
   
   void foo_host() {
     printf("%f\n", aaa);
   }

results in the following migrated SYCL\* code:

.. code-block:: cpp
   :linenos:

   /*
   DPCT1056:0: The use of aaa in device code was not detected. If this variable is
   also used in device code, you need to rewrite the code.
   */
   static const float aaa = 10.f;
   
   void foo_host() {
     printf("%f\n", aaa);
   }

> If the variable is only used in host code, ignore this warning,

Reviewer: ```suggestion
which is rewritten to:
1).  If the variable "aaa" in the code is only used in host code, just ignore the DPCT1056 warning

```

Author: done

> the code is rewritten to:

Reviewer: ```suggestion

```

Author: done

.. code-block:: cpp
   :linenos:

   static const float aaa = 10.f;
   
   void foo_host() {
     printf("%f\n", aaa);
   }

> If the variable is also used in device code,

Reviewer: ```suggestion
2) If the variable "aaa" is used in both host and device code: 

```

Author: done

If the variable is also used in device code, you need to migrate this code again
with ``--extra-arg="-xcuda"`` passed to the tool to explicitly specify the code language.

For example, this original CUDA\* code:

.. code-block:: cpp
   :linenos:

>    // dpct --out-root out test.c --extra-arg=-xc

Reviewer: L35 is not sample code, it's dpct command, please put it into desc.

Author: done

   
   // test.c 
   #include <stdio.h>
   #include <cuda_runtime.h>
   
   static __constant__ const float aaa = 10.f;
   
   void foo(){
     printf("%f\n", aaa);
   }

results in the following migrated SYCL\* code:

.. code-block:: cpp
   :linenos:

   /*
   DPCT1056:0: The use of aaa in device code was not detected. If this variable is
   also used in device code, you need to rewrite the code.
   */
   static const float aaa = 10.f;
   
   void foo(){
     printf("%f\n", aaa);
   }

which is rewritten to:

.. code-block:: cpp
   :linenos:

>    static const float aaa = 10.f;

Reviewer: Please give a ref code if this variable aaa is used in device code.

Author: done
